pallets/gear/src/lib.rs

```diff
@@ -1903,6 +1961,72 @@ pub mod pallet {
 
             Ok(())
         }
+
+        /// Process message queue
+        #[pallet::weight((
+            Weight::from_ref_time(BlockGasLimitOf::<T>::get()),
```

Comment:
can't we ran into some misbehaviour if our calculations of block gas limit of exceed current real amount due to our rounding on division?

And why not 0 at all?
Nobody will pay fee for it, so we can set 0 at the beginning and provide real amount in DispatchResult always

Reply:
Ok, I guess we can set the weigh of this extrinsic to 0. After all, it has a `DispatchClass::Mandatory` dispatch class and shouldn't interfere with the normal extrinsics whose total weight is determined by the `NORMAL_DISPATCH_RATIO`. The actual weight also doesn't really matter because this extrinsic is guaranteed to run last in a block, and always be included in a block (regardless of it's actual weight, as a mandatory dispatch).